File: Practica_1/src/gestorAplicacion/Lugar.java
package gestorAplicacion;

import java.io.Serializable;

/*Clase lugar hereda de servicio y se usa para crear los diferentes lugares ofrecidos(habitaciones y zonas sociales) */
public class Lugar extends Servicio {
    int numero;
    String tipo;
    int capacidad;

    // METODOS

    /*
     * Metodo encargado de consultar el valor de una habitacion segun el tipo que es
     * ingresado como parametro
     */
    public static int valorSegunTipo(String tipo) {
        if (tipo.equals("Habitación familiar")) {
            return 100000;
        } else if (tipo.equals("Habitación doble")) {
            return 80000;
        } else if (tipo.equals("Habitación individual")) {
            return 50000;
        } else if (tipo == "Terraza") {
            return 200000;
        } else if (tipo == "Piscina") {
            return 300000;
        } else if (tipo == "Salon") {
            return 500000;
        } else {
            return -1;
        }
    }

    /*
     * Metodo toString imprime toda la impormacion importante de la habitacion sobre
     * la que se ejecute el metodo
     */
    public String toString() {
        return "Habitacion: " + numero + " " + tipo + " con capacidad para " + capacidad + " personas";
    }

	// CONSTRUCTORES

    /*
     * Constructor para habitaciones
     * tipos: Habitación familiar, Habitación doble, Habitación individual
     */
    public Lugar(int numero, String tipo, int capacidad) {
        super("Habitacion " + tipo, valorSegunTipo(tipo));
        this.numero = numero;
        this.tipo = tipo;
        this.capacidad = capacidad;
    }

    /*
     * Constructor para Salas de eventos
     * tipos: Terraza, Piscina, Salon
     */

    public Lugar(String tipo) {
        super(tipo, valorSegunTipo(tipo));
        this.tipo = tipo;
    }

    // GETTERS AND SETTERS
    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public int getCapacidad() {
        return capacidad;
    }

    public void setCapacidad(int capacidad) {
        this.capacidad = capacidad;
    }

}

File: Python/gestorAplicacion/reservacion/Lugar.py
from Servicio import Servicio


class Lugar(Servicio):
    
    def __init__(self,nombre,descripcion,valor,numero,capacidad):
        valor=valor
        super().__init__(nombre, valor, descripcion)
        self._capacidad=capacidad
        self._numero=numero

        #El valor no se ingresa porque es calculado mediante la funcion ValorSegunTipo y el numero de los lugares de eventos es 0 por defecto

    def valorSegunTipo(nombre):
        if nombre == "Habitacion Familiar":
            return 100000
        elif nombre == "Habitacion Doble":
            return 80000
        elif nombre == "Habitacion Individual":
            return 50000
        elif nombre == "Terraza":
            return 200000
        elif nombre == "Piscina":
            return 300000
        elif nombre == "Salon":
            return 500000
        else:
            return -1

    def toString(self):
        return "Habitacion: " + str(self._numero) + " " + str(super().getNombre()) + " con capacidad para " + str(self._capacidad) + " personas"
    

    def setNumero(self, numero):
        self._numero = numero

    def getNumero(self):
        return self._numero

    def setCapacidad(self, capacidad):
        self._capacidad = capacidad

    def getCapacidad(self):
        return self._capacidad